File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/CBPacketEncryptStart.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.security.PublicRSA;
import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;

import java.io.IOException;

public class CBPacketEncryptStart extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 5;
    
    public CBPacketEncryptStart(){
        super(PACKET_ID);
    }
    
    
    public PublicRSA publicServerKey;
    
    public CBPacketEncryptStart(PublicRSA publicServerKey){
        this();
        this.publicServerKey = publicServerKey;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.write(publicServerKey.getKey().getEncoded());
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        publicServerKey = new PublicRSA(stream.readAllBytes());
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/CBPacketEncryptStart.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.security.PublicRSA
import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class CBPacketEncryptStart() : IPacket<PacketHandler?>(PACKET_ID) {
    var publicServerKey: PublicRSA? = null

    constructor(publicServerKey: PublicRSA?) : this() {
        this.publicServerKey = publicServerKey
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.write(publicServerKey!!.key!!.encoded)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        publicServerKey = PublicRSA(stream!!.readAllBytes())
    }

    companion object {
        const val PACKET_ID = 5
    }
}
